### PR TITLE
Add lint check to build.sh script

### PR DESCRIPTION
**Pull request by dsimansk**
## Description

@rhuss per one of our discussion on PR review.

## Changes

* Add `golangci-lint run` to default build flow
* Remove our `run_go_tool` in favour of `knative.dev/hack`

/cc @rhuss

### CODE REVIEW

File: hack/build.sh [hunk 1]
```diff
@@ -98,6 +100,9 @@ codegen() {
   # Format source code and cleanup imports
   source_format
 
+  # Lint source code
+  (( ! IS_PROW )) && source_lint
```
Comment (dsimansk) [inline hack/build.sh:104]: I've added skip on CI, since that check is part of GH actions. Any objections?

File: hack/build.sh [hunk 2]
```diff
@@ -110,30 +115,19 @@ go_fmt() {
   find $(echo $source_dirs) -name "*.go" -print0 | xargs -0 gofmt -s -w
 }
 
-# Run a go tool, get it first if necessary.
-run_go_tool() {
```
Comment (rhuss) [inline hack/build.sh:114]: This function has been removed but is still referenced below ?

Comment (dsimansk) [reply hack/build.sh:114]: I've opted for the common one from hack repo, https://github.com/knative/hack/blob/main/library.sh#L489-L514.

**Review by rhuss — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**